Adding reset_kc to setup_kernel

danielfrg opened the pull request:
Matching what async_setup_kernel does.

MSeal commented on nbclient/client.py:
```diff
@@ -411,6 +411,8 @@ def setup_kernel(self, **kwargs):
         When control returns from the yield it stops the client's zmq channels, and shuts
         down the kernel.
         """
+        reset_kc = kwargs.pop('reset_kc', False)
```
I'd rename this to `cleanup_kc` and change `def setup_kernel(self, **kwargs):` to `def setup_kernel(self, cleanup_kc=False, **kwargs):`

MSeal replied:
Ahh I see you just copied the async kernel implementation -- we should probably clean that up as well to make it clearer how it works, but I can also see the argument for keeping the current convention to avoid breaking people.

danielfrg commented:
I will add a test soon, I edited this directly on GitHub 🙈

Let me know what you prefer in terms of name/functionality.
My opinion: I agree that `cleanup_kc` is better.

I can also clean up `async_setup_kernel` as part of this, just let me know.

If you want to maintain the current convention I guess the default should be `cleanup_kc =True` by default, thats what happens now without the flag and seams reasonable to be because it's a contextmanager.

MSeal commented:
@davidbrochart what do you think about changing `reset_kc` to `cleanup_kc`?

davidbrochart commented:
Yes it is more coherent to change it to `cleanup_kc`. It's relatively new so I think we can afford that change.

danielfrg commented:
I cleaned changed `reset_kc` to `cleanup_kc` on the `*setup_kernel` functions and added a `test_cleanup_kernel_client` test.

Cleaned up a bit `reset_kc` on `execute`.

I think this should make it more coherent.
